src/Stratis.Bitcoin.Features.BlockStore/ProvenHeadersBlockStoreBehavior.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Base;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.P2P.Protocol;
+using Stratis.Bitcoin.P2P.Protocol.Payloads;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.BlockStore
+{
+    /// <inheritdoc />
+    public class ProvenHeadersBlockStoreBehavior : BlockStoreBehavior
+    {
+        private readonly Network network;
+
+        /// <summary>
+        /// Gets or sets the height from which start serving Proven Headers, if > 0.
+        /// </summary>
+        public int AnnounceProvenHeadersFromHeight { get; set; } = 0;
+
+        public ProvenHeadersBlockStoreBehavior(Network network, ConcurrentChain chain, IChainState chainState, ILoggerFactory loggerFactory, IConsensusManager consensusManager)
+            : base(chain, chainState, loggerFactory, consensusManager)
+        {
+            this.network = Guard.NotNull(network, nameof(network));
+        }
+
+        /// <inheritdoc />
+        protected async override Task ProcessMessageAsync(INetworkPeer peer, IncomingMessage message)
+        {
+            switch (message.Message.Payload)
+            {
+                case SendProvenHeadersPayload sendProvenHeadersPayload:
+                    await ProcessSendProvenHeadersPayload(sendProvenHeadersPayload);
+                    break;
+                default:
+                    await base.ProcessMessageAsync(peer, message).ConfigureAwait(false); ;
+                    break;
+            }
+        }
+
+        private Task ProcessSendProvenHeadersPayload(SendProvenHeadersPayload sendProvenHeadersPayload)
+        {
+            this.PreferHeaders = true;
```

Comment:
I am not sure this flag is part of PH, its about a node saying it preferes to receive headers, we should create a new flag if anything PreferProvenHeaders maybe?
But we dont use that because we default to anyway to prefer headers so its actually obsoulete

Reply:
I didn't created PreferProvenHeaders because PreferHeaders was enough to communicate that you prefer headers over Inv.
If you think it's obsolete then we should remove it completely from the project?
Or just leave it now then think about what to do later

Reply:
ok yeah leave it now but you dont need to set it do you? (ah yeah maybe you need ok)